src/Compilers/CSharp/Portable/Symbols/Source/SourceLocalSymbol.cs

```diff
@@ -60,8 +60,7 @@ internal class SourceLocalSymbol : LocalSymbol
             this.binder = binder;
             this._containingSymbol = containingSymbol;
             this._identifierToken = identifierToken;
-            this._refKind = refKind;
-            this._typeSyntax = typeSyntax;
+            this._typeSyntax = allowRefKind ? typeSyntax.SkipRef(out this._refKind) : typeSyntax;
```

Comment:
I think we should always SkipRef here. Symbol should not be associated with ref regardless of context.